test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -528,5 +528,78 @@ public void MatrixFactorizationBackCompat()
             // Negative example (i.e., examples can not be found in dataMatrix) is close to 0.15 (specified by s.C = 0.15 in the trainer).
             CompareNumbersWithTolerance(0.141411, testResults[1].Score, digitsOfPrecision: 5);
         }
+
+        [MatrixFactorizationFact]
+        public void OneClassMatrixFactorizationWithUnseenRow()
```

Comment:
>OneClassMatrixFactorizationWithUnseenRow [](start = 20, length = 40)

Right now you limits for columns are 2 and rows is 3.

Your sample have 2 columns and 2 rows.
Problem in code was related to border cases and triggered when we provide data which doesn't get to the limits.
Can we test fully? Right now you data doesn't reach max rows, can we test what would happen if you also don't reach max columns?

Reply:
Sure. Please take a look at the visualization of the latest training matrix below.

---
In reply to: [256537572](https://github.com/dotnet/machinelearning/pull/2525#discussion_r256537572) [](ancestors = 256537572)

Reply:
In the same spirit, it might be worth it to test an empty `IDataView` for training. 

---
In reply to: [256600662](https://github.com/dotnet/machinelearning/pull/2525#discussion_r256600662) [](ancestors = 256600662,256537572)